sophie-kaleba/assignment1/Anagram.java

```diff
@@ -1,86 +1,90 @@
- * - make the algorithm able to handle anagrams of sentences, where each word in the resulting sentence is an anagram of one of the wod in the original sentence
+ * Options:
+ * - handle both case sensitive and case insensitive anagrams
+ * - handle anagrams of sentences (each word = anagram of one of the word of the 1st sentence)
  */
 public class Anagram {
-	
+
 	/**
-	* Check if the two arguments given as parameter are anagrams. Handle both case sensitive and case insensitive comparisons.
-	* example: areAnagrams(listen, silent, false) returns True, areAnagrams(Listen, silent, true) returns False, areAnagrams(apple, pabble, true) returns False
-	* @param element1 - the 1st element
-	* @param element2 - the 2d element
-	* @param caseSenstive - true if the letter case is important ("listen" != "Listen")
-	* @return true if the 2 elements are anagrams, false otherwise
-	*
-	*/
+	 * Check if the two arguments given as parameter are anagrams. 
+	 * Handle both case sensitive and case insensitive comparisons.
+	 * example: areAnagrams(listen, silent, false) returns True, 
+	 * example: areAnagrams(Listen, silent, true) returns False
+	 * example: areAnagrams(apple, pabble, true) returns False
+	 * @param element1 - the 1st element
+	 * @param element2 - the 2d element
+	 * @param caseSenstive - true if the letter case is important ("listen" != "Listen")
+	 * @return true if the 2 elements are anagrams, false otherwise
+	 *
+	 */
 	public static boolean areAnagrams(String element1, String element2, boolean caseSensitive) {
-	    /* the element can be a sentence, split it according the standard control characters*/
+		/* the element can be a sentence, split it according the standard control characters*/
 		String[] splittedElement1 = element1.split(",|\\;| |\\:|\\.");
 		String[] splittedElement2 = element2.split(",|\\;| |\\:|\\.");
-		
+
 		/* if the element returned by split contains at least 2 words, this is a sentence */
 		if (splittedElement1.length>1 || splittedElement2.length>1 ) {
-		    return Anagram.AreAnagramsSentences(splittedElement1, splittedElement2);
+			return Anagram.AreAnagramsSentences(splittedElement1, splittedElement2, caseSensitive);
 		}
 		return Anagram.AreAnagramsWords(splittedElement1[0], splittedElement2[0], caseSensitive);
 	}
 
-	
+
 	/**
 	 * compares 2 sentences and check if they are anagrams
 	 * @param sentence1 - the 1st sentence
 	 * @param sentence2 - the 2d sentence
 	 * @return true if the 2 sentences are anagrams, false otherwise
 	 */
-	public static boolean AreAnagramsSentences(String[] sentence1, String[] sentence2) {
-		List<String> s1 = Arrays.asList(sentence1);
-		List<String> s2 = Arrays.asList(sentence2);
-		
+	public static boolean AreAnagramsSentences(String[] sentence1, String[] sentence2, boolean caseSensitive) {
+
 		/* must contains the same amount of words to be anagrams */
-		if (s1.size() == s2.size()) {
+		if (sentence1.length == sentence2.length) {
```

Comment:
Optional: It might be slightly easier to read if you avoid nesting, especially for simple checks.

What I mean is:

```java
  if (sentence1.length != sencent2.length) {
    return false;
  }

  for (String word : sentence1) {
    ...
  }
  return false;
```